python-spec/src/somacore/types.py

```diff
@@ -75,3 +76,12 @@ def is_slice_of(__obj: object, __typ: Type[_T]) -> TypeGuard[Slice[_T]]:
         and (__obj.stop is None or isinstance(__obj.stop, __typ))
         and (__obj.step is None or isinstance(__obj.step, __typ))
     )
+
+
+class ContextBase(Protocol):
+    """A protocol for a context manager that can be used as a base class.
+    If a threadpool is specified as part of the context, it will be used by the
+    implementer. Otherwise, the implementer will use its own threadpool.
```

Comment:
"it will be used by experiment queries", maybe? the implementer would be the one providing it